Package.swift

```diff
@@ -7,13 +7,20 @@ let package = Package(
     products: [
         .executable(name: "BasicDifferentiation", targets: ["BasicDifferentiation"]),
         .executable(name: "BasicGradientDescent", targets: ["BasicGradientDescent"]),
-        .executable(name: "CustomDerivatives", targets: ["CustomDerivatives"])
-	],     
-	targets: [
-        .executableTarget(name: "BasicDifferentiation", path: "BasicDifferentiation"),
-        .executableTarget(name: "BasicGradientDescent", path: "BasicGradientDescent"),
-        .executableTarget(name: "CustomDerivatives", path: "CustomDerivatives")
-	]
+        .executable(name: "CustomDerivatives", targets: ["CustomDerivatives"]),
+    ],
+    dependencies: [
+        .package(url: "https://github.com/apple/swift-format.git", .upToNextMajor(from: "509.0.0"))
```

Comment:
Why do we need `swift-format` as a dependency? Do we get easy access to a formatting plugin as a result?

Reply:
Oh that's a good call. I started out wanting to run it by using the plugin but in the end I'm using swift-format installed through brew since there's a bug in either command plugins or the swift-format plugin where I can't pass it a configuration file (https://github.com/apple/swift-format/issues/668) 
We can move to use the plugin and then also run the CI on linux once that is solved. But for now running on macOS and installing through brew was easier. I'll remove the dependency for now.